packages/@webex/plugin-meetings/src/meeting/voicea-meeting.ts

```diff
@@ -91,31 +61,51 @@ export const processNewCaptions = ({data, meeting}) => {
       transcriptData,
       csisKey: key,
     });
+    const {speakerId} = speaker;
+    const interimId = `${transcriptId}_${speakerId}`;
 
     if (needsCaching) {
       transcriptData.speakerProxy[key] = speaker;
     }
-    const {speakerId} = speaker;
-    const interimId = `${transcriptId}_${speakerId}`;
+
     const captionData = {
       id: interimId,
       isFinal: data.isFinal,
       translations: value.translations,
       text: value.text,
-      currentCaptionLanguage: value.currentSpokenLanguage,
+      currentCaptionLanguage:
+        meeting.transcription?.languageOptions?.currentCaptionLanguage ||
+        value.currentSpokenLanguage,
+      currentSpokenLanguage:
+        meeting.transcription?.languageOptions?.currentSpokenLanguage ||
+        data.transcripts[0]?.transcript_language_code,
       timestamp: value?.timestamp,
       speaker,
     };
 
-    const interimTranscriptIndex = transcriptData.captions.findIndex(
-      (transcript) => transcript.id === interimId
-    );
+    if (!data.isFinal) {
+      const interimTranscriptIndex = transcriptData.captions.findIndex(
+        (transcript) => transcript.id === interimId
+      );
 
-    if (interimTranscriptIndex !== -1) {
-      transcriptData.captions.splice(interimTranscriptIndex, 1);
-    }
+      if (interimTranscriptIndex !== -1) {
+        transcriptData.captions.splice(interimTranscriptIndex, 1);
+      }
 
-    interimTranscriptionIds.push(interimId);
+      interimTranscriptionIds.push(interimId);
+    } else {
+      transcriptData.interimCaptions[transcriptId].forEach((innerInterimId) => {
+        const interimTranscriptIndex = transcriptData.captions.findIndex(
+          (transcript) => transcript.id === innerInterimId
+        );
+
+        if (interimTranscriptIndex !== -1) {
+          transcriptData.captions.splice(interimTranscriptIndex, 1);
+        }
+      });
+      delete transcriptData.interimCaptions[transcriptId];
+      captionData.id = transcriptId;
```

Comment:
Would it be better to put the logic to find and remove the interimTranscript data into a separate function and call it here?

Reply:
Decided not to move forward after 1:1 Webex discussion that having a separate method just for splicing alone would also complicate it or would render no difference.